molecule/driver/openstackdriver.py

```diff
@@ -134,15 +134,23 @@ def up(self, no_provision=True):
                     security_groups=instance.get('security_groups', []))
                 self._reset_known_host_key(server['interface_ip'])
                 instance['created'] = True
+                instance['reachable'] = False
                 num_retries = 0
-                while not self._check_ssh_availability(
-                        server['interface_ip'],
-                        instance['sshuser'],
-                        timeout=6,
-                        sshkey_filename=self._get_keyfile(
-                        )) or num_retries == 5:
+                while num_retries < 5:
                     util.print_info('\t Waiting for ssh availability...')
                     num_retries += 1
+                    if self._check_ssh_availability(
+                            server['interface_ip'],
+                            instance['sshuser'],
+                            timeout=6,
+                            sshkey_filename=self._get_keyfile()):
+                        instance['reachable'] = True
+                        break
+                if not instance['reachable']:
```

Comment:
It seems a bit redundant to me to have the reachable flag. The num retries reaching a max threshold would already be a clear indicator of not being reachable, no?

I do like the add in of the print_error and sys.exit because that behavior needed to be added.